components/opponent/wikis/trackmania/opponent_display_custom.lua

```diff
@@ -0,0 +1,198 @@
+---
+-- @Liquipedia
+-- wiki=trackmania
+-- page=Module:OpponentDisplay/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Class = require('Module:Class')
+local DisplayUtil = require('Module:DisplayUtil')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Table = require('Module:Table')
+
+local Opponent = Lua.import('Module:Opponent', {requireDevIfEnabled = true})
+local OpponentDisplay = Lua.import('Module:OpponentDisplay', {requireDevIfEnabled = true})
+local PlayerDisplay = Lua.import('Module:Player/Display', {requireDevIfEnabled = true})
+
+local OpponentDisplayCustom = Table.deepCopy(OpponentDisplay)
+
+local SCORE_STATUS = 'S'
+local NO_SCORE = -1
+local ZERO_SCORE = 0
+
+--[[
+Display component for an opponent entry appearing in a bracket match.
+]]
+OpponentDisplayCustom.BracketOpponentEntry = Class.new(
+	function(self, opponent, options)
+		self.content = mw.html.create('div'):addClass('brkts-opponent-entry-left')
+
+		if opponent.type == Opponent.team then
+			self:createTeam(opponent.template or 'tbd', options)
+		elseif opponent.type == Opponent.solo or opponent.type == Opponent.duo then
+			self:createPlayers(opponent)
+		elseif opponent.type == Opponent.literal then
+			self:createLiteral(opponent.name or '')
+		end
+
+		self.root = mw.html.create('div'):addClass('brkts-opponent-entry')
+			:node(self.content)
+	end
+)
+
+OpponentDisplayCustom.BracketOpponentEntry.createTeam = OpponentDisplay.BracketOpponentEntry.createTeam
```

Comment:
Shouldn't be needed (deep copied at L20)

Reply:
same here as metioned in the other comment

Reply:
https://github.com/Liquipedia/Lua-Modules/pull/2262/commits/d41cb109c0ba4f4355e4fa10da9c51c1394d6913